enhancements/windows-containers/windows-machine-config-operator.md

```diff
@@ -0,0 +1,165 @@
+title: windows-machine-config-operator
+authors:
+  - "@ravisantoshgudimetla"
+  - "@aravindhp"
+reviewers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+  - "@openshift-team-cloud"
+  - "@openshift-team-mco"
+approvers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+creation-date: 2020-05-28
+last-updated: 2020-05-28
+status: implementable
+replaces:
+  - "/enhancements/windows-containers/ansible-dev-preview.md"
+---
+
+# Windows Machine Config Operator
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The intent of this enhancement is to allow a cluster administrator to add a
+Windows compute node with a prescribed configuration to an installer provisioned
+OpenShift 4.6 cluster on AWS, Azure and vSphere,configured with Hybrid OVN
+networking, as a day 2 operation and enable scheduling of Windows workloads.
+
+## Motivation
+
+The main motivation behind this enhancement is to satisfy customer
+requirement of being able to run Windows workloads on OpenShift clusters.
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Provide workflows for adding Windows instances provisioned using the Machine
+  API to OpenShift clusters.
+* Perform all the required steps within the node for it to be an OpenShift
+  compute node
+
+### Non-Goals
+
+As part of this enhancement we do not plan to support:
+* Installing the container runtime in the Windows node
+* Upgrades (This will be addressed in a separate enhancement)
+  * Windows operating system upgrades
+  * Node management like k8s compute component upgrades, reboots and node
+    draining
+* OpenShift Builds
+* OpenShift DeploymentConfigs
+
+## Proposal
+
+The two main pieces that will allow us to achieve our goals is the Windows
+Machine Config operator (WMCO) and a Windows Machine Config Bootstrapper (WMCB)
+binary. The operator will collect information from the cluster and transfer it
+to the Windows node. The WMCB will then perform the necessary steps on the node
+for it to join the cluster. [WMCB](https://github.com/openshift/windows-machine-config-bootstrapper)
+has already been developed as part of the Windows Containers dev preview.
+
+### Justification
+
+The reason we are using a WMCB binary rather than a container image is
+that all Windows container images contains a Windows kernel and Red Hat has a
+policy to not ship 3rd party kernels for support reasons.
+
+### Design Details
+
+The operator image will be packaged with the following binaries:
+* Kubelet
+* CNI plugins
+* Hybrid-overlay
+* WMCB
+* Kube-proxy
+
+WMCO will be published to OperatorHub as a Red Hat operator. The way the cluster
+admin will enable Windows workloads on their cluster would be to first install
```

Comment:
this section could also be reframed into steps or bullet points to make it easier to understand.